src/main/java/org/reaktivity/nukleus/socks/internal/stream/AbstractStreamProcessor.java

```diff
@@ -0,0 +1,383 @@
+        stream.accept(abort.typeId(), abort.buffer(), abort.offset(), abort.sizeof());
+    }
+
+    protected void doEnd(
+        MessageConsumer stream,
+        long streamId)
+    {
+        final EndFW end =
+            context.endRW
+                .wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+                .streamId(streamId)
+                .build();
+        stream.accept(end.typeId(), end.buffer(), end.offset(), end.sizeof());
+    }
+
+    protected void doWindow(
+        final MessageConsumer throttle,
+        final long throttleId,
+        final int credit,
+        final int padding)
+    {
+        final WindowFW window = context.windowRW
+            .wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+            .streamId(throttleId)
+            .credit(credit)
+            .padding(padding)
+            .build();
+        throttle.accept(window.typeId(), window.buffer(), window.offset(), window.sizeof());
+    }
+
+    protected void doReset(
+        final MessageConsumer throttle,
+        final long throttleId)
+    {
+        if (slotIndex != NO_SLOT)
+        {
+            // Can safely release the buffer
+            context.bufferPool.release(slotIndex);
+            slotWriteOffset = 0;
+            slotIndex = NO_SLOT;
+        }
+        final ResetFW reset =
+            context.resetRW
+                .wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+                .streamId(throttleId)
+                .build();
+        throttle.accept(reset.typeId(), reset.buffer(), reset.offset(), reset.sizeof());
+    }
+
+    protected void handleFragmentedData(
+        DataFW from,
+        FragmentedFlyweight to,
+        FragmentedHandler handler,
+        MessageConsumer throttle,
+        long throttleStreamId)
+    {
+        OctetsFW payload = from.payload();
+        DirectBuffer buffer = payload.buffer();
+        int limit = payload.limit();
+        int offset = payload.offset();
+        int size = limit - offset;
+        // Fragmented writes might have already occurred
+        if (slotIndex != NO_SLOT)
+        {
+            MutableDirectBuffer acceptBuffer = context.bufferPool.buffer(slotIndex, slotWriteOffset);
+            acceptBuffer.putBytes(0, buffer, offset, size);
+            slotWriteOffset += size;                                  // New starting point is moved to the end of the buffer
+            buffer = context.bufferPool.buffer(slotIndex);            // Try to decode from the beginning of the buffer
+            offset = 0;
+            limit = slotWriteOffset;
+        }
+        FragmentedFlyweight.ReadState fragmentationState = to.canWrap(buffer, offset, limit);
+        if (fragmentationState == FragmentedFlyweight.ReadState.FULL) // one negotiation request frame is in the buffer
+        {
+            try
+            {
+                handler.handle(to, buffer, offset, limit);
+            }
+            catch (Exception e)
```

Comment:
For performance reasons we avoid try catch unless we are calling external methods (as in TCP nukleus for example).  Looking at your FragmentedHandler implementations I do not believe any of them would throw exceptions anyway, so as it you can just remove the try catch. Any unexpected runtime exceptions should always be allowed to bubble up to the main reactor processor loop so it can generate appropriate diagnostics for internal errors (i.e. code bugs).

Reply:
I had done that to catch a NPE that was thrown when resolveTarget would return `null`. I am thinking to change the return type of the handling methods to `boolean` and return `false` when the `handleFragmentedData` and the other 2 methods should cleanup the buffer and reset, such as the case with `null`. Would this work ?